shared/auth/entitlement.go

```diff
@@ -0,0 +1,376 @@
+package auth
+
+import (
+	"fmt"
+
+	"github.com/canonical/lxd/shared"
+)
+
+// Entitlement represents a permission that can be applied to an entity.
+type Entitlement string
```

Comment:
@markylaing you mentioned the other day the reason these are in shared package is because they will/can be used by the `lxc` tool to validate valid entitlements before submitting them to the server. This is unusual to perform low level config setting/value validation on the client before doing it on the server, and the reason we don't generally do this is that the `lxc` tool is used to communicate with servers of differing versions.

So in the future when we no doubt alter the entitlement list, the old `lxc` clients won't know about the new entitlements and wont allow them to be used even if the server supports them.

With that in mind we may need to (later) extend the metadata api to provide a list of entitelements supported by the server, and so perhaps these don't need to be in shared package after all.

Thoughts?

Reply:
This is a very good point. I agree these should move back to `lxd/auth`. 

What about `entity.Type`? These are must less likely to change, but it is conceivable that we would add new ones (`TypeDeployment`, `TypeShape` etc.). I guess in that case the user will need to update their CLI.